sunpy/net/vso/vso.py

```diff
@@ -92,15 +92,23 @@ def check_connection(url):
     except (socket.error, socket.timeout, HTTPError, URLError) as e:
         warnings.warn(
             "Connection failed with error {}. \n Retrying with different url and port.".format(e))
+        return None
 
 
 def get_online_vso_url(api, url, port):
-    if api is None and (url is None or port is None):
-        for mirror in DEFAULT_URL_PORT:
-            if check_connection(mirror['url']):
-                api = zeep.Client(mirror['url'], port_name=mirror['port'])
-                api.set_ns_prefix('VSO', 'http://virtualsolar.org/VSO/VSOi')
-                return api
+    if isinstance(api, zeep.client.Client):
+        if check_connection(api.wsdl.location):
```

Comment:
I wonder if this is needed, I would assume that you would be unable to create a `zeep.Client` object if the WSDL was offline. I think we can just do
```
if isinstance(api, zeep.client.Client):
    return api
```

here.

Reply:
You're right. It's not possible to create a `zeep.Client` object when you're offline. I thought it would be safe to check WSDL status before returning API (just in case zeep.Client object was created a while ago)

Reply:
I think we are probably safe to remove the extra check, I would imagine that anyone setting the api by themselves would be doing so for weird edge cases anyway.